cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+
+1. **Azure AD Workload Identity mutating admission webhook (AzWI mutating admission webhook)** is a pod on the Kubernetes cluster that intercepts the pod creation request and injects the signed service account token into the pod. This signed service account token is used by the Azure provider to authenticate with Azure resources.
+
+
+## Objectives
+
+> **Issue Reference: https://github.com/radius-project/radius/issues/7308**
+
+### Goals
+
+* Radius users can configure Azure provider to use Workload Identity for authentication.
+* Workload Identity can be configured via interactive experience.
+* Workload Identity can be configured manually.
+* Radius users can deploy and manage Azure resources without using service principal credentials.
+
+### Non-goals
+
+* Azure Managed Identity support
+* AWS Workload Identity (IRSA) support
+
+### User scenarios
+
+#### User story 1: As an infrastructure operator, I can configure the Radius Azure Provider to use Workload Identity for authentication via interactive experience.
+
+1. Install the `rad` CLI.
+
+1. run `rad init --full` to begin the interactive experience
+    ```
+    rad init --full
+    ```
+
+1. Enter the environment name, namespace and select "yes" to add a cloud provider and select "Azure" to add an Azure cloud provider.
+    ```
+    Select your cloud provider:
+    > 1. Azure
+      2. AWS
+      3. [back]
+    ```
+
+1. Enter or use the Azure subscription and resource group name. Select  "Workload Identity" for the Azure cloud provider.
+    ```
+    Select the identity for the Azure cloud provider 
+    1. Service principal                               
+    > 2. Workload Identity    
+    ```
+
+1. Enter the Entra ID Application Client ID.
+    ```
+    Please follow the guidance at aka.ms/rad-workload-identity to set up workload identity for Radius.
+
+    Enter the Client ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Client ID:
+    ```
+
+Radius is now configured to use Workload Identity for authentication with Azure.
+
+#### User story 2: As an infrastructure operator, I can configure the Radius Azure Provider to use Workload Identity for authentication manually.
+
+1. Install the `rad` CLI.
+
+1. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.azureWorkloadIdentity.enabled=true
+    ```
+
+1. Create a Radius environment and add an Azure cloud provider.
+    ```
+    rad group create default
+    rad env create default
+    rad env update default --azure-subscription-id <subscription-id> --azure-resource-group <resource-group> 
+    ```
+
+1. Register the Entra ID Application using `rad credential register azure`.
+    ```
+    rad credential register azure --client-id <client-id>
```

Comment:
What is the expected behavior here if workload identity isn't enabled in step 2 before this?